pennylane/ops/op_math/controlled.py

```diff
@@ -562,17 +552,19 @@ def matrix(self, wire_order=None):
 
         shape = qml.math.shape(base_matrix)
         if len(shape) == 3:  # stack if batching
-            canonical_matrix = qml.math.stack(
+            return qml.math.stack(
                 [qml.math.block_diag([left_pad, _U, right_pad]) for _U in base_matrix]
             )
-        else:
-            canonical_matrix = qmlmath.block_diag([left_pad, base_matrix, right_pad])
 
-        if wire_order is None or self.wires == Wires(wire_order):
-            return qml.math.expand_matrix(
-                canonical_matrix, wires=self.active_wires, wire_order=self.wires
-            )
+        return qmlmath.block_diag([left_pad, base_matrix, right_pad])
 
+    def matrix(self, wire_order=None):
+        if self.compute_matrix is not Operator.compute_matrix:
+            canonical_matrix = self.compute_matrix(*self.data)
```

Comment:
This is not elegant but it seems to be a better alternative than every custom controlled op with a custom `compute_matrix` having to override `matrix` with duplicated code.

Reply:
I actually think it's pretty solid, gets the job done. are we sure it never needs hyperparameters?

Reply:
The canonical matrix does not take into account wires, which is what the hyperparameters of a controlled operation contains, so I'm pretty certain that it will not need them.

Reply:
I am more worried about things like `self.target_wires` or `self.control_wires` that are used in today's `matrix()` function, but I suppose we'll learn quickly if it ever fails in the future

Reply:
For controlled operations, the `compute_matrix()` method does not consider wires at all. The wire mapping is handled in `matrix()`